### ISSUE TITLE
[REQUEST] webp support

### ISSUE THREAD

**Issue by Lunatik00** [keras-team/keras#17013]
**System information**.

TensorFlow version (you are using): 2.10
Are you willing to contribute it (Yes/No) : Can't

**Describe the feature and the current behavior/state**.

I am reading a set of images using  tf.keras.utils.image_dataset_from_directory( ), the problem arises because I have been using webp format and I was surprised when it wasn't amongst the image formats supported by this function (or any other function I saw on the tensorflow API), the expected behavior was to use .webp as any other format.

A function to decode .webp already exists in tensorflow_io (not in tensorflow.io) https://www.tensorflow.org/io/api_docs/python/tfio/image/decode_webp

**Will this change the current api? How?**

Maybe it will add some extra functions that could be needed to use this format

**Who will benefit from this feature?**

Anyone that wants to use webp, the reason I am using it is because it can save images in a loseless format while doing a compression equivalent to jpeg (around the same filesize with less data lost) and in lose format with a much bigger compression, it saves a lot of space and it was compatible with everything else I was using, I can just read the image using cv2 and then pass it to tensorflow, but that adds a lot of read time and increases the ram requirements, i did a test for open, resize and save the same images using opencv and tensorflow and the results in time difference was massive (2 seconds in tensorflow using the tf.keras.utils.image_dataset_from_directory( ) function, several minutes in opencv because the code was sequential.

TLDR, it will help everyone that chooses to use the webp format for any particular reason alongside keras/tensorflow increasing compatibility.

**Comment by Lunatik00** [comment]
Hi, the main problem is in the tf.decode_image function

 what I tested:

- add ".webp" to the ALLOWLIST_FORMATS variable in keras.utils.image_dataset
- changing the decode image to decode_webp reads the image, but it reads rgba, the image is rgb causing other issues in the long run, that function is in keras.utils.image_dataset and is called load_image(), it ended up as tfio.image.decode_webp(img,channels=num_channels) and not using the expand_animations=False kwarg
- Given the lack of the channels kwarg I added that to the decode_webp function so it read rgba if 4, converts rgba to rgb if 3 and converts rgba to rgb to grayscale if it is 1 (I didn't found a rgba to grayscale function), but that lacks the 0 option that reads the channels in the image because I have no idea how to implement that, so I didn't pushed the change. It was in tensorflow_io.python.ops.image_ops

That gave me the dataset and it was usable, but as you can see that is not an appropriate solution for the root of the problem, the root of the problem is the decode_image function, I have posted this on the tensorflow repository and I hope it is solved by them, specially because they have a decode function that they can just copy for the heavy part of the work, then the final change would be to add .webp to the allowed list, a solution from the keras side would be to import tfio if available, to check if the image is webp in the load_image function and to load it using tfio.decode_webp (i don't know how to do the check when it is in bytes format), although I still think the real problem is in the tf.decode_image function

**Comment by haifeng-jin** [comment]
Summary: support `.webp` format of images in `tf.keras.utils.image_dataset_from_directory( )`.

Maybe we can mark it as contribution welcome?

**Comment by fchollet** [comment]
If you (@Lunatik00) or another contributor is able to open a functional & performant PR to add `webp` support, we will review and merge.

Otherwise, one workaround you can try is to first convert your webps to pngs via a very small shell script. You can use the dwebp utility, e.g. `dwebp file.webp -o file.png`.

**Comment by Lunatik00** [comment]
For now it is with workarounds, the function that must be modified first is the tf.decode_image and at least I don't completely understand it for now, as I said, I understand how to make it work with my particular problem but that is not a general solution, since I have negative time available for now I am using other formats, if I have the time I will try to make the decode image function to work with webp, but I only opened this because the TF team said it was a keras issue at first and I hadn't gone deep searching the root problem, currently they understood that is the decode_image function ( https://github.com/tensorflow/tensorflow/issues/57647#issuecomment-1252753294_  ) and that when it is solved there is just adding it to the ALLOWLIST_FORMATS from the keras team. As a comment, I am kinda amazed that google doesn't push this format more, it is very good

**Comment by SuryanarayanaY** [comment]
Hello, Thank you for reporting an issue.

We're currently in the process of migrating the new Keras 3 code base from keras-team/keras-core to keras-team/keras.
Consequently, This issue may not be relevant to the Keras 3 code base . After the migration is successfully completed, feel free to reopen this Issue at keras-team/keras if you believe it remains relevant to the Keras 3 code base. If instead this Issue is a bug or security issue in legacy tf.keras, you can instead report a new issue at [keras-team/tf-keras](https://github.com/keras-team/tf-keras/issues), which hosts the TensorFlow-only, legacy version of Keras.

To know more about Keras 3, please read https://keras.io/keras_core/announcement/
